diff -X .gitignore -Naurb ././01-getting-started/2017.journal ././02-getting-data-in/2017.journal
--- ././01-getting-started/2017.journal
+++ ././02-getting-data-in/2017.journal
@@ -1,31 +1,9 @@
 ;; This journal is for the current year
+include commodities.journal
 
 ;; Opening balances
 2017/01/01 opening balances
   assets:Lloyds:current    = £100.00
   equity:opening balances
 
-2017-01-31  End-of-month balance
-  assets:Lloyds:current  =   £840.61
-  income:employer        £-800.11 
-  expenses:unknown
-
-2017-02-28  End-of-month balance
-  assets:Lloyds:current  =  £1626.75
-  income:employer        £-900.22
-  expenses:unknown
-
-2017-03-31  End-of-month balance
-  assets:Lloyds:current  =  £2618.31
-  income:employer        £-1093.72 
-  expenses:unknown
-
-2017-04-30  End-of-month balance
-  assets:Lloyds:current  =  £3322.48
-  income:employer        £-800.72 
-  expenses:unknown
-
-2017-05-31  End-of-month balance
-  assets:Lloyds:current  =  £4058.83
-  income:employer        £-903.52 
-  expenses:unknown
+include ./import/lloyds/journal/99966633_20171223_1844.journal
diff -X .gitignore -Naurb ././01-getting-started/README.md ././02-getting-data-in/README.md
--- ././01-getting-started/README.md
+++ ././02-getting-data-in/README.md
@@ -1,3 +1,11 @@
 ## What's here
 
-Simple empty journal without any transactions
+- One year of transactions from the single bank account, imported from CSV files, with automatic data scrubbing and CSV-to-journal conversion
+
+- Auto-generated reports:
+
+    - balance sheet
+
+    - cash flow
+
+    - income/expenses
diff -X .gitignore -Naurb ././01-getting-started/commodities.journal ././02-getting-data-in/commodities.journal
--- ././01-getting-started/commodities.journal
+++ ././02-getting-data-in/commodities.journal
@@ -0,0 +1 @@
+commodity £1000.00
diff -X .gitignore -Naurb ././01-getting-started/export/export.hs ././02-getting-data-in/export/export.hs
--- ././01-getting-started/export/export.hs
+++ ././02-getting-data-in/export/export.hs
@@ -58,7 +58,9 @@
 -----------------------------------------
 -- Extra dependencies of the import files
 -----------------------------------------
-extraDeps file = []
+extraDeps file
+  | "//lloyds//*.journal" ?== file   = ["lloyds.rules"]
+  | otherwise = []
 
 -----------------------------------------------
 -- Extra inputs to be fed to conversion scripts
@@ -128,10 +130,10 @@
   (opening_balances "//*") %> generate_opening_balances flags year_inputs
 
   -- Enumerate directories with auto-generated cleaned csv files
-  [ ] |%> in2csv
+  [ "//import/lloyds/csv/*.csv" ] |%> in2csv
 
   -- Enumerate directories with auto-generated journals
-  [ ] |%> csv2journal
+  [ "//import/lloyds/journal/*.journal" ] |%> csv2journal
 
 -------------------------------------
 -- Implementations of the build rules
diff -X .gitignore -Naurb ././01-getting-started/import/lloyds/csv2journal ././02-getting-data-in/import/lloyds/csv2journal
--- ././01-getting-started/import/lloyds/csv2journal
+++ ././02-getting-data-in/import/lloyds/csv2journal
@@ -0,0 +1,2 @@
+#!/bin/bash
+hledger print --rules-file lloyds.rules -f "$1"
diff -X .gitignore -Naurb ././01-getting-started/import/lloyds/in/99966633_20171223_1844.csv ././02-getting-data-in/import/lloyds/in/99966633_20171223_1844.csv
--- ././01-getting-started/import/lloyds/in/99966633_20171223_1844.csv
+++ ././02-getting-data-in/import/lloyds/in/99966633_20171223_1844.csv
@@ -0,0 +1,21 @@
+Transaction Date,Transaction Type,Sort Code,Account Number,Transaction Description,Debit Amount,Credit Amount,Balance,
+25/05/2017,BGC,'12-34-56,99966633,EMPLOYER INC,,903.52,4058.83
+15/05/2017,BP,'12-34-56,99966633,OASIS COFFEE ,2.76,,3155.31
+05/05/2017,DEB,'12-34-56,99966633,WAITROSE,64.41,,3158.07
+01/05/2017,BP,'12-34-56,99966633,AVIVA,100,,3222.48
+25/04/2017,BGC,'12-34-56,99966633,EMPLOYER INC,,800.72,3322.48
+18/04/2017,BP,'12-34-56,99966633,OASIS COFFEE ,2.76,,2521.76
+07/04/2017,BP,'12-34-56,99966633,OASIS COFFEE ,2.76,,2524.52
+07/04/2017,DEB,'12-34-56,99966633,WAITROSE,92.24,,2527.28
+01/04/2017,,'12-34-56,99966633,INTEREST (NET) ,,1.21,2619.52
+31/03/2017,BGC,'12-34-56,99966633,HSBC,100,,2618.31
+25/03/2017,BGC,'12-34-56,99966633,EMPLOYER INC,,1093.72,2718.31
+12/03/2017,BP,'12-34-56,99966633,OASIS COFFEE ,2.16,,1624.59
+25/02/2017,BGC,'12-34-56,99966633,EMPLOYER INC,,900.22,1626.75
+10/02/2017,BP,'12-34-56,99966633,OASIS COFFEE ,2.76,,726.53
+05/02/2017,DEB,'12-34-56,99966633,WAITROSE,111.32,,729.29
+25/01/2017,BGC,'12-34-56,99966633,EMPLOYER INC,,800.11,840.61
+15/01/2017,BP,'12-34-56,99966633,OASIS COFFEE ,2.76,,40.50
+10/01/2017,BP,'12-34-56,99966633,OASIS COFFEE ,2.76,,43.26
+09/01/2017,DEB,'12-34-56,99966633,WAITROSE,51.22,,46.02
+05/01/2017,BP,'12-34-56,99966633,OASIS COFFEE ,2.76,,97.24
diff -X .gitignore -Naurb ././01-getting-started/import/lloyds/in2csv ././02-getting-data-in/import/lloyds/in2csv
--- ././01-getting-started/import/lloyds/in2csv
+++ ././02-getting-data-in/import/lloyds/in2csv
@@ -0,0 +1,2 @@
+#!/bin/bash
+sed -e 's/  +/ /g; s/,99966633,/,assets:Lloyds:current,/' < "$1"
diff -X .gitignore -Naurb ././01-getting-started/import/lloyds/lloyds.rules ././02-getting-data-in/import/lloyds/lloyds.rules
--- ././01-getting-started/import/lloyds/lloyds.rules
+++ ././02-getting-data-in/import/lloyds/lloyds.rules
@@ -0,0 +1,14 @@
+fields date,code,sortcode,account1,description,amount1-out,amount1-in,balance1
+skip 1
+
+date-format %d/%m/%Y
+currency1 £
+account2  expenses:unknown
+
+if
+INTEREST
+  account2   income:interest
+
+if
+EMPLOYER INC
+  account2   income:employer
