diff -X .gitignore -Naurb ././01-getting-started/2017.journal ././02-getting-data-in/2017.journal
--- ././01-getting-started/2017.journal
+++ ././02-getting-data-in/2017.journal
@@ -1 +1,8 @@
 ;; This journal is for the current year
+
+;; Opening balances
+2017/01/01 opening balances
+  assets:Lloyds:current    = £100.00
+  equity:opening balances
+  
+include ./import/lloyds/journal/99966633_20171223_1844.journal
diff -X .gitignore -Naurb ././01-getting-started/README.md ././02-getting-data-in/README.md
--- ././01-getting-started/README.md
+++ ././02-getting-data-in/README.md
@@ -1,3 +1,11 @@
 ## What's here
 
-Simple empty journal without any transactions
+- One year of transactions from the single bank account, imported from CSV files, with automatic data scrubbing and CSV-to-journal conversion
+
+- Auto-generated reports:
+
+    - balance sheet
+
+    - cash flow
+
+    - income/expenses
diff -X .gitignore -Naurb ././01-getting-started/export/export.hs ././02-getting-data-in/export/export.hs
--- ././01-getting-started/export/export.hs
+++ ././02-getting-data-in/export/export.hs
@@ -57,7 +57,9 @@
 -----------------------------------------
 -- Extra dependencies of the import files
 -----------------------------------------
-extraDeps file = []
+extraDeps file
+  | "//lloyds//*.journal" ?== file   = ["lloyds.rules"]
+  | otherwise = []
 
 -----------------------------------------------
 -- Extra inputs to be fed to conversion scripts
@@ -118,10 +120,10 @@
   (opening_balances "//*") %> generate_opening_balances flags year_inputs
 
   -- Enumerate directories with auto-generated cleaned csv files
-  [ ] |%> in2csv
+  [ "//import/lloyds/csv/*.csv" ] |%> in2csv
 
   -- Enumerate directories with auto-generated journals
-  [ ] |%> csv2journal
+  [ "//import/lloyds/journal/*.journal" ] |%> csv2journal
 
 -------------------------------------
 -- Implementations of the build rules
diff -X .gitignore -Naurb ././01-getting-started/import/lloyds/csv2journal ././02-getting-data-in/import/lloyds/csv2journal
--- ././01-getting-started/import/lloyds/csv2journal
+++ ././02-getting-data-in/import/lloyds/csv2journal
@@ -0,0 +1,2 @@
+#!/bin/bash
+hledger print --rules-file lloyds.rules -f "$1"
diff -X .gitignore -Naurb ././01-getting-started/import/lloyds/in/99966633_20171223_1844.csv ././02-getting-data-in/import/lloyds/in/99966633_20171223_1844.csv
--- ././01-getting-started/import/lloyds/in/99966633_20171223_1844.csv
+++ ././02-getting-data-in/import/lloyds/in/99966633_20171223_1844.csv
@@ -0,0 +1,7 @@
+Transaction Date,Transaction Type,Sort Code,Account Number,Transaction Description,Debit Amount,Credit Amount,Balance,
+01/05/2017,BP,'12-34-56,99966633,AVIVA,100,,986.93
+07/04/2017,BP,'12-34-56,99966633,OASIS COFFEE ,2.76,,1086.93
+07/04/2017,DEB,'12-34-56,99966633,WAITROSE,5.24,,1089.69
+01/04/2017,,'12-34-56,99966633,INTEREST (NET) ,,1.21,1094.93
+31/03/2017,BGC,'12-34-56,99966633,HSBC,100,,1093.72
+30/03/2017,BGC,'12-34-56,99966633,EMPLOYER INC,,1093.72,1193.72
diff -X .gitignore -Naurb ././01-getting-started/import/lloyds/in2csv ././02-getting-data-in/import/lloyds/in2csv
--- ././01-getting-started/import/lloyds/in2csv
+++ ././02-getting-data-in/import/lloyds/in2csv
@@ -0,0 +1,2 @@
+#!/bin/bash
+sed -e 's/  +/ /g; s/,99966633,/,assets:Lloyds:current,/' < "$1"
diff -X .gitignore -Naurb ././01-getting-started/import/lloyds/lloyds.rules ././02-getting-data-in/import/lloyds/lloyds.rules
--- ././01-getting-started/import/lloyds/lloyds.rules
+++ ././02-getting-data-in/import/lloyds/lloyds.rules
@@ -0,0 +1,14 @@
+fields date,code,sortcode,account1,description,amount1-out,amount1-in,balance1
+skip 1
+
+date-format %d/%m/%Y
+currency1 £
+account2  expenses:unknown
+
+if
+INTEREST
+  account2   income:interest
+
+if
+EMPLOYER INC
+  account2   income:employer
