diff -X .gitignore -Naurb ././01-getting-started/2017.journal ././02-getting-data-in/2017.journal
--- ././01-getting-started/2017.journal	2019-02-06 23:33:02.000000000 +0000
+++ ././02-getting-data-in/2017.journal	2019-02-06 23:33:18.000000000 +0000
@@ -1 +1,8 @@
 ;; This journal is for the current year
+
+;; Opening balances
+2017/01/01 opening balances
+  assets:Lloyds:current    = £100
+  equity:opening balances
+  
+!include ./import/lloyds/journal/99966633_20171223_1844.journal
diff -X .gitignore -Naurb ././01-getting-started/export/2017-accounts.txt ././02-getting-data-in/export/2017-accounts.txt
--- ././01-getting-started/export/2017-accounts.txt	2019-02-08 23:22:10.772250104 +0000
+++ ././02-getting-data-in/export/2017-accounts.txt	2019-02-08 23:22:11.808268102 +0000
@@ -0,0 +1,5 @@
+assets:Lloyds:current
+equity:opening balances
+expenses
+income:employer
+income:interest
diff -X .gitignore -Naurb ././01-getting-started/export/2017-all.journal ././02-getting-data-in/export/2017-all.journal
--- ././01-getting-started/export/2017-all.journal	2019-02-08 23:13:43.835727031 +0000
+++ ././02-getting-data-in/export/2017-all.journal	2019-02-06 23:33:18.000000000 +0000
@@ -0,0 +1,20 @@
+2017/01/01 opening balances
+    assets:Lloyds:current       = £100
+    equity:opening balances
+
+2017/03/31 (BGC) EMPLOYER INC
+    assets:Lloyds:current         £993.72 = £1093.72
+    income:employer              £-993.72
+
+2017/04/01 INTEREST (NET)
+    assets:Lloyds:current           £1.21 = £1094.93
+    income:interest                £-1.21
+
+2017/04/07 (DEB) WAITROSE
+    assets:Lloyds:current          £-5.24 = £1089.69
+    expenses                        £5.24
+
+2017/04/07 (BP) OASIS COFFEE
+    assets:Lloyds:current          £-2.76 = £1086.93
+    expenses                        £2.76
+
diff -X .gitignore -Naurb ././01-getting-started/export/2017-balance-sheet.txt ././02-getting-data-in/export/2017-balance-sheet.txt
--- ././01-getting-started/export/2017-balance-sheet.txt	2019-02-08 23:13:43.823726830 +0000
+++ ././02-getting-data-in/export/2017-balance-sheet.txt	2019-02-06 23:33:18.000000000 +0000
@@ -1,15 +1,18 @@
-Balance Sheet -
+Balance Sheet 2017/01/01-2017/04/07
 
-             ||  
-=============++==
+             || 2017/04/07 
+=============++============
  Assets      ||  
--------------++--
--------------++--
-             ||  
-=============++==
+-------------++------------
+ assets      ||   £1086.93 
+   Lloyds    ||   £1086.93 
+     current ||   £1086.93 
+-------------++------------
+             ||   £1086.93 
+=============++============
  Liabilities ||  
--------------++--
--------------++--
+-------------++------------
+-------------++------------
              ||  
-=============++==
- Net:        ||  
+=============++============
+ Net:        ||   £1086.93 
diff -X .gitignore -Naurb ././01-getting-started/export/2017-cash-flow.txt ././02-getting-data-in/export/2017-cash-flow.txt
--- ././01-getting-started/export/2017-cash-flow.txt	2019-02-08 23:22:10.780250243 +0000
+++ ././02-getting-data-in/export/2017-cash-flow.txt	2019-02-08 23:22:11.792267824 +0000
@@ -1,8 +1,9 @@
-Cashflow Statement -
+Cashflow Statement 2017/01/01-2017/04/07
 
-            || - 
-============++===
+                       || 2017/01/01-2017/04/07 
+=======================++=======================
  Cash flows ||   
-------------++---
-------------++---
-            ||   
+-----------------------++-----------------------
+ assets:Lloyds:current ||               £986.93 
+-----------------------++-----------------------
+                       ||               £986.93 
diff -X .gitignore -Naurb ././01-getting-started/export/2017-closing.journal ././02-getting-data-in/export/2017-closing.journal
--- ././01-getting-started/export/2017-closing.journal	2019-02-08 23:22:10.780250243 +0000
+++ ././02-getting-data-in/export/2017-closing.journal	2019-02-08 23:22:11.804268033 +0000
@@ -1,3 +1,4 @@
 2017/12/31 closing balances
-    equity:closing balances               0
+    assets:Lloyds:current         £-1086.93 = £0
+    equity:closing balances
 
diff -X .gitignore -Naurb ././01-getting-started/export/2017-income-expenses.txt ././02-getting-data-in/export/2017-income-expenses.txt
--- ././01-getting-started/export/2017-income-expenses.txt	2019-02-08 23:13:43.819726764 +0000
+++ ././02-getting-data-in/export/2017-income-expenses.txt	2019-02-06 23:33:18.000000000 +0000
@@ -1,15 +1,18 @@
-Income Statement -
+Income Statement 2017/01/01-2017/04/07
 
-          || - 
-==========++===
+                 || 2017/01/01-2017/04/07 
+=================++=======================
  Revenues ||   
-----------++---
-----------++---
-          ||   
-==========++===
+-----------------++-----------------------
+ income:employer ||               £993.72 
+ income:interest ||                 £1.21 
+-----------------++-----------------------
+                 ||               £994.93 
+=================++=======================
  Expenses ||   
-----------++---
-----------++---
-          ||   
-==========++===
- Net:     ||   
+-----------------++-----------------------
+ expenses        ||                 £8.00 
+-----------------++-----------------------
+                 ||                 £8.00 
+=================++=======================
+ Net:            ||               £986.93 
diff -X .gitignore -Naurb ././01-getting-started/export/export.hs ././02-getting-data-in/export/export.hs
--- ././01-getting-started/export/export.hs	2019-02-08 23:13:27.495453072 +0000
+++ ././02-getting-data-in/export/export.hs	2019-02-08 23:34:16.060650546 +0000
@@ -147,7 +147,10 @@
   ------------    
   -- Importing
   ------------
-  let extraDeps file = []
+  let extraDeps file
+        | "//lloyds/csv/*.csv" ?== file    = ["./lloyds2csv.pl"]
+        | "//lloyds//*.journal" ?== file   = ["lloyds.rules"]
+        | otherwise = []
 
   let extraInputs file = []
 
@@ -179,7 +182,9 @@
         (Stdout output) <- cmd (Cwd source_dir) "./csv2journal" [makeRelative source_dir input]
         writeFileChanged out output
 
-  return ()
+  "//import//lloyds/csv/*.csv" %> in2csv
+  "//import//lloyds/journal/*.journal" %> csv2journal
+  
   where
     -- To get included files, look for '!include'. Note that we cant use "hledger files", as
     -- some of the requested includes might be generated by this file and might not exist yet.
diff -X .gitignore -Naurb ././01-getting-started/import/lloyds/csv/99966633_20171223_1844.csv ././02-getting-data-in/import/lloyds/csv/99966633_20171223_1844.csv
--- ././01-getting-started/import/lloyds/csv/99966633_20171223_1844.csv	1970-01-01 01:00:00.000000000 +0100
+++ ././02-getting-data-in/import/lloyds/csv/99966633_20171223_1844.csv	2019-02-06 23:33:18.000000000 +0000
@@ -0,0 +1,5 @@
+Transaction Date,Transaction Type,Sort Code,Account Number,Transaction Description,Debit Amount,Credit Amount,Balance
+07/04/2017,BP,'12-34-56,assets:Lloyds:current,OASIS COFFEE ,2.76,,1086.93
+07/04/2017,DEB,'12-34-56,assets:Lloyds:current,WAITROSE,5.24,,1089.69
+01/04/2017,,'12-34-56,assets:Lloyds:current,INTEREST (NET) ,,1.21,1094.93
+31/03/2017,BGC,'12-34-56,assets:Lloyds:current,EMPLOYER INC,,993.72,1093.72
diff -X .gitignore -Naurb ././01-getting-started/import/lloyds/csv2journal ././02-getting-data-in/import/lloyds/csv2journal
--- ././01-getting-started/import/lloyds/csv2journal	1970-01-01 01:00:00.000000000 +0100
+++ ././02-getting-data-in/import/lloyds/csv2journal	2019-02-08 23:31:08.373468499 +0000
@@ -0,0 +1,2 @@
+#!/bin/bash
+hledger print --rules-file lloyds.rules -f "$1"
diff -X .gitignore -Naurb ././01-getting-started/import/lloyds/in/99966633_20171223_1844.csv ././02-getting-data-in/import/lloyds/in/99966633_20171223_1844.csv
--- ././01-getting-started/import/lloyds/in/99966633_20171223_1844.csv	1970-01-01 01:00:00.000000000 +0100
+++ ././02-getting-data-in/import/lloyds/in/99966633_20171223_1844.csv	2019-02-06 23:33:18.000000000 +0000
@@ -0,0 +1,5 @@
+Transaction Date,Transaction Type,Sort Code,Account Number,Transaction Description,Debit Amount,Credit Amount,Balance,
+07/04/2017,BP,'12-34-56,99966633,OASIS COFFEE ,2.76,,1086.93
+07/04/2017,DEB,'12-34-56,99966633,WAITROSE,5.24,,1089.69
+01/04/2017,,'12-34-56,99966633,INTEREST (NET) ,,1.21,1094.93
+31/03/2017,BGC,'12-34-56,99966633,EMPLOYER INC,,993.72,1093.72
diff -X .gitignore -Naurb ././01-getting-started/import/lloyds/in2csv ././02-getting-data-in/import/lloyds/in2csv
--- ././01-getting-started/import/lloyds/in2csv	1970-01-01 01:00:00.000000000 +0100
+++ ././02-getting-data-in/import/lloyds/in2csv	2019-02-08 23:30:26.276753002 +0000
@@ -0,0 +1,2 @@
+#!/bin/bash
+./lloyds2csv.pl < "$1"
diff -X .gitignore -Naurb ././01-getting-started/import/lloyds/journal/99966633_20171223_1844.journal ././02-getting-data-in/import/lloyds/journal/99966633_20171223_1844.journal
--- ././01-getting-started/import/lloyds/journal/99966633_20171223_1844.journal	1970-01-01 01:00:00.000000000 +0100
+++ ././02-getting-data-in/import/lloyds/journal/99966633_20171223_1844.journal	2019-02-06 23:33:18.000000000 +0000
@@ -0,0 +1,16 @@
+2017/03/31 (BGC) EMPLOYER INC
+    assets:Lloyds:current         £993.72 = £1093.72
+    income:employer              £-993.72
+
+2017/04/01 INTEREST (NET)
+    assets:Lloyds:current           £1.21 = £1094.93
+    income:interest                £-1.21
+
+2017/04/07 (DEB) WAITROSE
+    assets:Lloyds:current          £-5.24 = £1089.69
+    expenses                        £5.24
+
+2017/04/07 (BP) OASIS COFFEE
+    assets:Lloyds:current          £-2.76 = £1086.93
+    expenses                        £2.76
+
diff -X .gitignore -Naurb ././01-getting-started/import/lloyds/lloyds2csv.pl ././02-getting-data-in/import/lloyds/lloyds2csv.pl
--- ././01-getting-started/import/lloyds/lloyds2csv.pl	1970-01-01 01:00:00.000000000 +0100
+++ ././02-getting-data-in/import/lloyds/lloyds2csv.pl	2019-02-06 23:33:18.000000000 +0000
@@ -0,0 +1,14 @@
+#!/usr/bin/perl
+while (<>) {
+  chomp;
+  s/\r$//;
+  # Remove quoted strings with commas
+  s/"([A-Z0-9]+),/"$1 /g;
+  s/"//g;
+  # Collapse spaces
+  s/  +/ /g;
+  @fields=split(/,/,$_);
+  # Clean up account names
+  $fields[3] =~ s/99966633/assets:Lloyds:current/;
+  print join(",",@fields)."\n";
+}
diff -X .gitignore -Naurb ././01-getting-started/import/lloyds/lloyds.rules ././02-getting-data-in/import/lloyds/lloyds.rules
--- ././01-getting-started/import/lloyds/lloyds.rules	1970-01-01 01:00:00.000000000 +0100
+++ ././02-getting-data-in/import/lloyds/lloyds.rules	2019-02-06 23:33:18.000000000 +0000
@@ -0,0 +1,16 @@
+fields date,code,sortcode,account1,description,amount-out,amount-in,balance
+skip 1
+
+date-format %d/%m/%Y
+currency £
+
+account1  assets:Lloyds:current
+account2  expenses
+
+if
+INTEREST
+  account2   income:interest
+
+if
+EMPLOYER INC
+  account2   income:employer
