diff -X .gitignore -Naurb ././02-getting-data-in/2014.journal ././03-getting-full-history/2014.journal
--- ././02-getting-data-in/2014.journal
+++ ././03-getting-full-history/2014.journal
@@ -0,0 +1,7 @@
+;; Opening balances
+2014/01/01 opening balances
+  assets:Lloyds:current    = £100.00
+  assets:cash              = £150.00
+  equity:opening balances
+
+include ./import/lloyds/journal/99966633_20171224_2041.journal
diff -X .gitignore -Naurb ././02-getting-data-in/2015.journal ././03-getting-full-history/2015.journal
--- ././02-getting-data-in/2015.journal
+++ ././03-getting-full-history/2015.journal
@@ -0,0 +1,2 @@
+include ./export/2015-opening.journal
+include ./import/lloyds/journal/99966633_20171224_2042.journal
diff -X .gitignore -Naurb ././02-getting-data-in/2016.journal ././03-getting-full-history/2016.journal
--- ././02-getting-data-in/2016.journal
+++ ././03-getting-full-history/2016.journal
@@ -0,0 +1,2 @@
+include ./export/2016-opening.journal
+include ./import/lloyds/journal/99966633_20171224_2043.journal
diff -X .gitignore -Naurb ././02-getting-data-in/2017.journal ././03-getting-full-history/2017.journal
--- ././02-getting-data-in/2017.journal
+++ ././03-getting-full-history/2017.journal
@@ -1,8 +1,4 @@
 ;; This journal is for the current year
 
-;; Opening balances
-2017/01/01 opening balances
-  assets:Lloyds:current    = £100.00
-  equity:opening balances
-  
+include ./export/2017-opening.journal
 include ./import/lloyds/journal/99966633_20171223_1844.journal
diff -X .gitignore -Naurb ././02-getting-data-in/README.md ././03-getting-full-history/README.md
--- ././02-getting-data-in/README.md
+++ ././03-getting-full-history/README.md
@@ -1,6 +1,6 @@
 ## What's here
 
-- One year of transactions from the single bank account, imported from CSV files, with automatic data scrubbing and CSV-to-journal conversion
+- Several years of transactions from the single bank account, imported from CSV files, with automatic data scrubbing and CSV-to-journal conversion
 
 - Auto-generated reports:
 
diff -X .gitignore -Naurb ././02-getting-data-in/all.journal ././03-getting-full-history/all.journal
--- ././02-getting-data-in/all.journal
+++ ././03-getting-full-history/all.journal
@@ -1 +1,7 @@
+include 2014.journal
+include ./export/2014-closing.journal
+include 2015.journal
+include ./export/2015-closing.journal
+include 2016.journal
+include ./export/2016-closing.journal
 include 2017.journal
diff -X .gitignore -Naurb ././02-getting-data-in/export/export.hs ././03-getting-full-history/export/export.hs
--- ././02-getting-data-in/export/export.hs
+++ ././03-getting-full-history/export/export.hs
@@ -18,7 +18,7 @@
 -- 3. Name of the hledger binary
 -- 4. Which accounts to include into opening/closing balances
 --
-defaultFirstYear     = 2017 :: Int
+defaultFirstYear     = 2014 :: Int
 defaultCurrentYear   = 2017
 defaultBaseDir       = ".."
 defaultHledgerBinary = "hledger"
diff -X .gitignore -Naurb ././02-getting-data-in/import/lloyds/in/99966633_20171224_2041.csv ././03-getting-full-history/import/lloyds/in/99966633_20171224_2041.csv
--- ././02-getting-data-in/import/lloyds/in/99966633_20171224_2041.csv
+++ ././03-getting-full-history/import/lloyds/in/99966633_20171224_2041.csv
@@ -0,0 +1,5 @@
+Transaction Date,Transaction Type,Sort Code,Account Number,Transaction Description,Debit Amount,Credit Amount,Balance,
+01/05/2014,BP,'12-34-56,99966633,AVIVA,100,,600.00
+07/04/2014,DEB,'12-34-56,99966633,WAITROSE,73.72,,700.00
+31/03/2014,BGC,'12-34-56,99966633,HSBC,100,,773.72
+30/03/2014,BGC,'12-34-56,99966633,EMPLOYER INC,,773.72,873.72
diff -X .gitignore -Naurb ././02-getting-data-in/import/lloyds/in/99966633_20171224_2042.csv ././03-getting-full-history/import/lloyds/in/99966633_20171224_2042.csv
--- ././02-getting-data-in/import/lloyds/in/99966633_20171224_2042.csv
+++ ././03-getting-full-history/import/lloyds/in/99966633_20171224_2042.csv
@@ -0,0 +1,6 @@
+Transaction Date,Transaction Type,Sort Code,Account Number,Transaction Description,Debit Amount,Credit Amount,Balance,
+01/05/2015,BP,'12-34-56,99966633,AVIVA,100,,650.00
+08/04/2015,BP,'12-34-56,99966633,OASIS COFFEE ,3.72,,750.00
+07/04/2015,DEB,'12-34-56,99966633,TRANSFER TO 12345678,500,,753.72
+31/03/2015,BGC,'12-34-56,99966633,HSBC,100,,1253.72
+30/03/2015,BGC,'12-34-56,99966633,EMPLOYER INC,,753.72,1353.72
diff -X .gitignore -Naurb ././02-getting-data-in/import/lloyds/in/99966633_20171224_2043.csv ././03-getting-full-history/import/lloyds/in/99966633_20171224_2043.csv
--- ././02-getting-data-in/import/lloyds/in/99966633_20171224_2043.csv
+++ ././03-getting-full-history/import/lloyds/in/99966633_20171224_2043.csv
@@ -0,0 +1,6 @@
+Transaction Date,Transaction Type,Sort Code,Account Number,Transaction Description,Debit Amount,Credit Amount,Balance,
+01/05/2016,BP,'12-34-56,99966633,AVIVA,100,,100.00
+09/04/2016,DEB,'12-34-56,99966633,TRANSFER TO 12345678,1000,,200.00
+07/04/2016,BP,'12-34-56,99966633,OASIS COFFEE ,3.72,,1200.00
+31/03/2016,BGC,'12-34-56,99966633,HSBC,100,,1203.72
+30/03/2016,BGC,'12-34-56,99966633,EMPLOYER INC,,653.72,1303.72
diff -X .gitignore -Naurb ././02-getting-data-in/import/lloyds/lloyds.rules ././03-getting-full-history/import/lloyds/lloyds.rules
--- ././02-getting-data-in/import/lloyds/lloyds.rules
+++ ././03-getting-full-history/import/lloyds/lloyds.rules
@@ -12,3 +12,7 @@
 if
 EMPLOYER INC
   account2   income:employer
+
+if
+TRANSFER TO 12345678
+  account2   assets:Lloyds:savings
