diff -X .gitignore -Naurb ././03-getting-full-history/2015.journal ././04-adding-more-accounts/2015.journal
--- ././03-getting-full-history/2015.journal
+++ ././04-adding-more-accounts/2015.journal
@@ -1,2 +1,3 @@
 include ./export/2015-opening.journal
 include ./import/lloyds/journal/99966633_20171224_2042.journal
+include ./import/lloyds/journal/12345678_20171225_0001.journal
diff -X .gitignore -Naurb ././03-getting-full-history/2016.journal ././04-adding-more-accounts/2016.journal
--- ././03-getting-full-history/2016.journal
+++ ././04-adding-more-accounts/2016.journal
@@ -1,2 +1,3 @@
 include ./export/2016-opening.journal
 include ./import/lloyds/journal/99966633_20171224_2043.journal
+include ./import/lloyds/journal/12345678_20171225_0002.journal
diff -X .gitignore -Naurb ././03-getting-full-history/README.md ././04-adding-more-accounts/README.md
--- ././03-getting-full-history/README.md
+++ ././04-adding-more-accounts/README.md
@@ -1,6 +1,6 @@
 ## What's here
 
-- Several years of transactions from the single bank account, imported from CSV files, with automatic data scrubbing and CSV-to-journal conversion
+- Several years of transactions from several bank account, imported from CSV files, with automatic data scrubbing and CSV-to-journal conversion
 
 - Auto-generated reports:
 
diff -X .gitignore -Naurb ././03-getting-full-history/import/lloyds/in/12345678_20171225_0001.csv ././04-adding-more-accounts/import/lloyds/in/12345678_20171225_0001.csv
--- ././03-getting-full-history/import/lloyds/in/12345678_20171225_0001.csv
+++ ././04-adding-more-accounts/import/lloyds/in/12345678_20171225_0001.csv
@@ -0,0 +1,2 @@
+Transaction Date,Transaction Type,Sort Code,Account Number,Transaction Description,Debit Amount,Credit Amount,Balance,
+07/04/2015,DEB,'12-34-56,12345678,TRANSFER FROM 99966633,,500,500.00
diff -X .gitignore -Naurb ././03-getting-full-history/import/lloyds/in/12345678_20171225_0002.csv ././04-adding-more-accounts/import/lloyds/in/12345678_20171225_0002.csv
--- ././03-getting-full-history/import/lloyds/in/12345678_20171225_0002.csv
+++ ././04-adding-more-accounts/import/lloyds/in/12345678_20171225_0002.csv
@@ -0,0 +1,2 @@
+Transaction Date,Transaction Type,Sort Code,Account Number,Transaction Description,Debit Amount,Credit Amount,Balance,
+09/04/2016,DEB,'12-34-56,12345678,TRANSFER FROM 99966633,,1000,1500.00
diff -X .gitignore -Naurb ././03-getting-full-history/import/lloyds/in2csv ././04-adding-more-accounts/import/lloyds/in2csv
--- ././03-getting-full-history/import/lloyds/in2csv
+++ ././04-adding-more-accounts/import/lloyds/in2csv
@@ -1,2 +1,2 @@
 #!/bin/bash
-sed -e 's/  +/ /g; s/,99966633,/,assets:Lloyds:current,/' < "$1"
+sed -e 's/  +/ /g; s/,99966633,/,assets:Lloyds:current,/; s/,12345678,/,assets:Lloyds:savings,/' < "$1"
diff -X .gitignore -Naurb ././03-getting-full-history/import/lloyds/lloyds.rules ././04-adding-more-accounts/import/lloyds/lloyds.rules
--- ././03-getting-full-history/import/lloyds/lloyds.rules
+++ ././04-adding-more-accounts/import/lloyds/lloyds.rules
@@ -14,5 +14,5 @@
   account2   income:employer
 
 if
-TRANSFER TO 12345678
-  account2   assets:Lloyds:savings
+TRANSFER
+  account2   assets:Lloyds:transfers
