diff -X .gitignore -Naurb ././04-adding-more-accounts/export/export.hs ././05-creating-csv-import-rules/export/export.hs
--- ././04-adding-more-accounts/export/export.hs
+++ ././05-creating-csv-import-rules/export/export.hs
@@ -29,6 +29,8 @@
 income_expenses   y = y++"-income-expenses.txt"
 balance_sheet     y = y++"-balance-sheet.txt"
 cash_flow         y = y++"-cash-flow.txt"
+accounts          y = y++"-accounts.txt"
+unknown           y = y++"-unknown.journal"
 -- which accounts to include in opening/closing reports
 open_close_account_query = "assets|liabilities|debts"
 closing_balances  y = y++"-closing.journal"
@@ -39,9 +41,11 @@
 --
 reports =
   concat [ [ transactions         (show y) | y <- all_years ]
+         , [ accounts             (show y) | y <- all_years ]
          , [ income_expenses      (show y) | y <- all_years ]
          , [ balance_sheet        (show y) | y <- all_years ]
          , [ cash_flow            (show y) | y <- all_years ]
+         , [ unknown              (show y) | y <- all_years ]
          , [ opening_balances     (show y) | y <- all_years, y/=first ]
          , [ closing_balances     (show y) | y <- all_years, y/=current ]
          ]
@@ -73,12 +77,16 @@
 
   (transactions "//*") %> hledger_process_year year_inputs ["print"]
 
+  (accounts "//*") %> hledger_process_year year_inputs ["accounts"]
+
   (income_expenses "//*") %> hledger_process_year year_inputs ["is","--flat"]
 
   (balance_sheet "//*") %> hledger_process_year year_inputs ["balancesheet"]
 
   (cash_flow "//*") %> hledger_process_year year_inputs ["cashflow","not:desc:(opening balances)"]
 
+  (unknown "//*") %> hledger_process_year year_inputs ["print", "unknown"]
+
   (closing_balances "//*") %> generate_closing_balances year_inputs
 
   (opening_balances "//*") %> generate_opening_balances year_inputs
diff -X .gitignore -Naurb ././04-adding-more-accounts/import/lloyds/lloyds.rules ././05-creating-csv-import-rules/import/lloyds/lloyds.rules
--- ././04-adding-more-accounts/import/lloyds/lloyds.rules
+++ ././05-creating-csv-import-rules/import/lloyds/lloyds.rules
@@ -15,3 +15,15 @@
 if
 TRANSFER
   account2   assets:Lloyds:transfers
+
+if
+WAITROSE
+  account2   expenses:groceries
+
+if
+OASIS COFFEE
+  account2   expenses:coffee
+
+if
+AVIVA
+  account2   assets:pension:aviva
diff -X .gitignore -Naurb ././04-adding-more-accounts/README.md ././05-creating-csv-import-rules/README.md
--- ././04-adding-more-accounts/README.md
+++ ././05-creating-csv-import-rules/README.md
@@ -2,6 +2,8 @@
 
 - Several years of transactions from several bank account, imported from CSV files, with automatic data scrubbing and CSV-to-journal conversion
 
+- CSV import rules file to classify expenses
+
 - Auto-generated reports:
 
     - balance sheet
