diff -X .gitignore -Naurb ././05-creating-csv-import-rules/README.md ././06-maintaining-lots-of-csv-rules/README.md
--- ././05-creating-csv-import-rules/README.md
+++ ././06-maintaining-lots-of-csv-rules/README.md
@@ -2,7 +2,7 @@
 
 - Several years of transactions from several bank account, imported from CSV files, with automatic data scrubbing and CSV-to-journal conversion
 
-- CSV import rules file to classify expenses
+- Compact rules file to classify expenses
 
 - Auto-generated reports:
 
diff -X .gitignore -Naurb ././05-creating-csv-import-rules/export/export.hs ././06-maintaining-lots-of-csv-rules/export/export.hs
--- ././05-creating-csv-import-rules/export/export.hs
+++ ././06-maintaining-lots-of-csv-rules/export/export.hs
@@ -62,7 +62,7 @@
 -- Extra dependencies of the import files
 -----------------------------------------
 extraDeps file
-  | "//lloyds//*.journal" ?== file   = ["lloyds.rules"]
+  | "//lloyds//*.journal" ?== file   = ["lloyds.rules", "rules.psv"]
   | otherwise = []
 
 -----------------------------------------------
diff -X .gitignore -Naurb ././05-creating-csv-import-rules/import/lloyds/lloyds.rules ././06-maintaining-lots-of-csv-rules/import/lloyds/lloyds.rules
--- ././05-creating-csv-import-rules/import/lloyds/lloyds.rules
+++ ././06-maintaining-lots-of-csv-rules/import/lloyds/lloyds.rules
@@ -5,26 +5,4 @@
 currency1 £
 account2  expenses:unknown
 
-if
-INTEREST
-  account2   income:interest
-
-if
-EMPLOYER INC
-  account2   income:employer
-
-if
-TRANSFER
-  account2   assets:Lloyds:transfers
-
-if
-WAITROSE
-  account2   expenses:groceries
-
-if
-OASIS COFFEE
-  account2   expenses:coffee
-
-if
-AVIVA
-  account2   assets:pension:aviva
+include rules.psv
diff -X .gitignore -Naurb ././05-creating-csv-import-rules/import/lloyds/rules.psv ././06-maintaining-lots-of-csv-rules/import/lloyds/rules.psv
--- ././05-creating-csv-import-rules/import/lloyds/rules.psv
+++ ././06-maintaining-lots-of-csv-rules/import/lloyds/rules.psv
@@ -0,0 +1,7 @@
+if|account2|comment
+INTEREST|income:interest|
+EMPLOYER INC|income:employer|
+TRANSFER|assets:Lloyds:transfers|
+WAITROSE|expenses:groceries|
+OASIS COFFEE|expenses:coffee|
+AVIVA|assets:pension:aviva|
