diff -X .gitignore -Naurb ././14-speeding-up/2014.journal ././15-manual-lots/2014.journal
--- ././14-speeding-up/2014.journal
+++ ././15-manual-lots/2014.journal
@@ -16,5 +16,19 @@
   assets:pension:aviva     = £102.34
   virtual:unrealized pnl
   
+;; Pension allowance tracking
+2014-04-05
+   (virtual:pension:allowance:2013/2014)    £4000           ; how much the allowance was
+   virtual:pension:allowance:2013/2014    = £0              ; how much of it is left
+   virtual:pension:inputs:2013/2014       £100.00           ; how much pension contributions were
+   virtual:pension:allowance:unused:2013/2014 - 2016/2017   ; remainder went to unused allowance
+
+;; Stock options tracking - 5 UNITS granted (to vest in two years), 0 vested
+2014/12/30 Stock options
+  virtual:stock options:granted
+  virtual:stock options:vesting:2016   5 UNITS
+
+P 2014/12/30 UNITS $708.75
+
 include ./import/lloyds/journal/99966633_20171224_2041.journal
 include ./export/2014-mortgage-interest.journal
diff -X .gitignore -Naurb ././14-speeding-up/2015.journal ././15-manual-lots/2015.journal
--- ././14-speeding-up/2015.journal
+++ ././15-manual-lots/2015.journal
@@ -5,6 +5,20 @@
   assets:pension:aviva     = £204.41
   virtual:unrealized pnl
   
+;; Pension allowance tracking
+2015-04-05
+   (virtual:pension:allowance:2014/2015)  £4000             ; how much the allowance was
+   virtual:pension:allowance:2014/2015    = £0              ; how much of it is left
+   virtual:pension:inputs:2014/2015       £100.00           ; how much pension contributions were
+   virtual:pension:allowance:unused:2014/2015 - 2017/2018   ; remainder went to unused allowance
+
+;; Stock options tracking - 10 UNITS granted (to vest in two years), 0 vested
+2015/12/30 Stock Options
+  virtual:stock options:granted
+  virtual:stock options:vesting:2017   10 UNITS
+
+P 2015/12/30 UNITS $654.77
+
 include ./import/lloyds/journal/99966633_20171224_2042.journal
 include ./import/lloyds/journal/12345678_20171225_0001.journal
 include ./export/2015-mortgage-interest.journal
diff -X .gitignore -Naurb ././14-speeding-up/2016.journal ././15-manual-lots/2016.journal
--- ././14-speeding-up/2016.journal
+++ ././15-manual-lots/2016.journal
@@ -5,6 +5,25 @@
   assets:pension:aviva     = £308.27
   virtual:unrealized pnl
   
+;; Pension allowance tracking
+2016-04-05
+   (virtual:pension:allowance:2015/2016)  £50             ; how much the allowance was
+   virtual:pension:allowance:2015/2016    = £0            ; how much of it is left
+   virtual:pension:inputs:2015/2016       £100.00         ; how much pension contributions were
+   virtual:pension:allowance:unused:2013/2014 - 2016/2017   -£50 ; we need to use £50 of 2013/2014 allowance
+   virtual:pension:allowance:unused:2015/2016 - 2018/2019   ; remainder went to unused allowance
+
+;; Stock options tracking - 20 UNITS granted (to vest in two years), year 2016 vested
+2016/12/30 Stock Options
+  virtual:stock options:granted
+  virtual:stock options:vesting:2018   20 UNITS
+
+2016/12/30 Stock Options
+  virtual:stock options:vested
+  virtual:stock options:vesting:2016   = 0 UNITS
+
+P 2016/12/30 UNITS $851.12
+
 include ./import/lloyds/journal/99966633_20171224_2043.journal
 include ./import/lloyds/journal/12345678_20171225_0002.journal
 include ./export/2016-mortgage-interest.journal
diff -X .gitignore -Naurb ././14-speeding-up/2017.journal ././15-manual-lots/2017.journal
--- ././14-speeding-up/2017.journal
+++ ././15-manual-lots/2017.journal
@@ -12,6 +12,28 @@
    (p60:tax paid)              -£2744.63
    (p60:national insurance)    -£2000.66
 
+;; Pension allowance tracking
+2017-04-05
+   (virtual:pension:allowance:2016/2017)  £40             ; how much the allowance was
+   virtual:pension:allowance:2016/2017    = £0            ; how much of it is left
+   virtual:pension:inputs:2016/2017       £100.00         ; how much pension contributions were
+   virtual:pension:allowance:unused:2014/2015 - 2017/2018   -£60 ; we need to use £60 of 2014/2015 allowance
+   virtual:pension:allowance:unused:2016/2017 - 2019/2020   ; remainder went to unused allowance
+
+; Under UK rules, this allowance should expire in 2018, but I am lazy and dont want to create 2018.journal to illustrate this, so lets expire it in 2017
+2017-04-01 Expired
+   (virtual:pension:allowance:unused:2013/2014 - 2016/2017)  = £0
+
+;; Stock options tracking - 25 UNITS granted (to vest in two years), year 2017 vested
+2017/12/30 Stock Options
+  virtual:stock options:granted
+  virtual:stock options:vesting:2019   25 UNITS
+
+2017/12/30 Stock Options
+  virtual:stock options:vested
+  virtual:stock options:vesting:2017   = 0 UNITS
+
+P 2017/12/30 UNITS $901.97
 
 include ./import/lloyds/journal/99966633_20171223_1844.journal
 include ./import/lloyds/journal/12345678_20171225_0003.journal
diff -X .gitignore -Naurb ././14-speeding-up/commodities.journal ././15-manual-lots/commodities.journal
--- ././14-speeding-up/commodities.journal
+++ ././15-manual-lots/commodities.journal
@@ -1,2 +1,3 @@
 commodity £1000.00
 commodity $1000.00
+commodity 1000. UNITS
diff -X .gitignore -Naurb ././14-speeding-up/export/export.hs ././15-manual-lots/export/export.hs
--- ././14-speeding-up/export/export.hs
+++ ././15-manual-lots/export/export.hs
@@ -50,6 +50,8 @@
 investments              = "investments.txt"
 mortgage_interest y = y++"-mortgage-interest.journal"
 tax_return y = printf "%d-%d-tax.txt" (y-1) y
+pension_contributions = "pension.txt"
+stock_options = "stock-options.txt"
 
 --
 -- Defining the full set of reports and journals to be generated
@@ -63,7 +65,7 @@
          , [ unknown              (show y) | y <- all_years ]
          , [ opening_balances     (show y) | y <- all_years, y/=first ]
          , [ closing_balances     (show y) | y <- all_years, y/=current ]
-         , [ investments ]
+         , [ investments, pension_contributions, stock_options ]
          , [ mortgage_interest    (show y) | y <- all_years ]
          , [ tax_return                 y  | y <- [2017..current] ]
          ]
@@ -159,7 +161,11 @@
   -- Enumerate directories with auto-generated journals
   [ "//import/lloyds/journal/*.journal" ] |%> csv2journal
 
-  ("//" ++ investments) %> generate_investments_report current year_inputs
+  ("//" ++ investments) %> with_all_investment_years current year_inputs "./investments.sh" ["none"]
+
+  ("//" ++ pension_contributions) %> with_all_investment_years current year_inputs "./pension.sh" ["none"]
+
+  ("//" ++ stock_options) %> with_all_investment_years current year_inputs "./stock-options.sh" ["none"]
 
   -- Mortgage interest transactions
   mortgage_interest "//*" %> generate_mortgage_interest year_inputs
@@ -243,11 +249,11 @@
 -------------------
 -- Helper functions
 -------------------
-generate_investments_report current year_inputs out = do
+with_all_investment_years current year_inputs prog args out = do
   deps <- mapM (year_inputs . show) (investment_years current)
   need (concat deps)
-  need [ "./investments.sh" ]
-  (Stdout output) <- cmd "./investments.sh"
+  need [prog]
+  (Stdout output) <- cmd prog args
   writeFileChanged out output
 
 generate_mortgage_interest year_inputs out = do
diff -X .gitignore -Naurb ././14-speeding-up/export/pension.sh ././15-manual-lots/export/pension.sh
--- ././14-speeding-up/export/pension.sh
+++ ././15-manual-lots/export/pension.sh
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+cd $(dirname $0)
+
+parallel -k :::: <<EOF
+echo "PENSION CONTRIBUTIONS"
+echo "====================="
+hledger -f ../all.journal balance "assets:pension:aviva" "virtual:pension:input" -b "2013-04-06" -p "every year" --transpose --depth 3
+echo
+
+echo "PENSION ALLOWANCE TRACKING"
+echo "=========================="
+hledger  -f ../all.journal balance virtual:pension -b '2013-04-06' -p 'every year' --no-elide --tree --cumulative -H
+echo
+
+echo "UNUSED PENSION ALLOWANCE"
+echo "========================"
+hledger  -f ../all.journal register virtual:pension:allowance:unused -b '2013-04-06' --depth 4
+EOF
diff -X .gitignore -Naurb ././14-speeding-up/export/stock-options.sh ././15-manual-lots/export/stock-options.sh
--- ././14-speeding-up/export/stock-options.sh
+++ ././15-manual-lots/export/stock-options.sh
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+cd $(dirname $0)
+
+parallel -k :::: <<EOF
+echo "END OF YEAR VALUATIONS"
+echo "======================"
+hledger balance 'stock options:vested' -f ../all.journal -Y -b2014 --cumulative --transpose --value=end,$ --valuechange --no-total
+echo
+
+echo "VESTING HISTORY"
+echo "==============="
+hledger balance 'stock options:vest' -f ../all.journal -Y -b2014 --cumulative
+echo
+EOF
